src/services/storage-cluster/allocator.ts

```diff
@@ -797,13 +797,7 @@ export class StorageClusterAllocator extends StorageCluster {
     this.wss.on('connection', (ws) => {
       let authenticated = false;
       let peerId: string;
-      ws.on('message', async (data) => {
-        let message: SocketMsg;
-        try {
-          message = JSON.parse(data.toString());
-        } catch {
-          return;
-        }
+      ws.on('message', async (message: SocketMsg) => {
```

Comment:
This type is incorrect. `data` is a `Buffer` or `string`. This should be inferred, right?
```suggestion
      ws.on('message', async (data: Buffer | string) => {
        let message: SocketMsg;
        try {
          message = JSON.parse(data.toString());
        } catch {
          return;
        }
```

Reply:
That is so weird. Maybe there is a type issue in the WebSocket lib you are using?